src/TicTacToeModel.java

```diff
@@ -0,0 +1,77 @@
+public class TicTacToeModel {
+    public enum Player {X, O, EMPTY}
+    private final Player[][] board = new Player[3][3];
+    private Player currentPlayer;
+
+    public TicTacToeModel() {
+        currentPlayer = Player.X;
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                board[i][j] = Player.EMPTY;
+            }
+        }
+    }
+
+    public void restartGame(){
+        currentPlayer = Player.X;
+        for (int i = 0; i < 3; i++) {
+            for (int j = 0; j < 3; j++) {
+                board[i][j] = Player.EMPTY;
+            }
+        }
+    }
+
+    public void checkBounds(int row, int col) {
+        if (row < 0 || row > 2 || col < 0 || col > 2) {
+            throw new IllegalArgumentException();
+        }
```

Comment:
Consider improving the exception handling by providing a more informative message, such as "Invalid board position: row and column must be between 0 and 2."